test/integ/pandas/test_arrow_pandas.py

```diff
@@ -436,6 +436,23 @@ def test_timestampntz(conn_cnx, scale):
         finish(conn, table)
 
 
+@pytest.mark.parametrize("timestamp_str", ["'1400-01-01 01:02:03.123456789'::timestamp as low_ts", "'9999-01-01 01:02:03.123456789789'::timestamp as high_ts"])
+def test_timestampntz_raises_overflow(conn_cnx, timestamp_str):
+    with conn_cnx() as conn:
+        r = conn.cursor().execute(f"select {timestamp_str}")
+        with pytest.raises(OverflowError) as exec_info:
+            r.fetch_arrow_all()
+        assert "overflows int64 range." in str(exec_info.value)
```

Comment:
[pytest.raises](https://docs.pytest.org/en/6.2.x/reference.html#pytest.raises) actually accepts a regex to match against the Exception. You could use that to remove the follow up assert statement, like
```suggestion
        with pytest.raises(OverflowError, match=r"overflows int64 range."):
            r.fetch_arrow_all()
```